agent/util-scripts/pbench-tool-meister-start

```diff
@@ -282,34 +284,77 @@ def main(argv):
         logger.exception("failed to load tool group data")
         return 1
 
+    # See if anybody told us to use certain options with SSH commands.
+    ssh_opts = os.environ.get("ssh_opts", "")
+
     try:
         benchmark_run_dir = os.environ["benchmark_run_dir"]
         hostname = os.environ["_pbench_hostname"]
         full_hostname = os.environ["_pbench_full_hostname"]
     except Exception:
         logger.exception("failed to fetch parameters from the environment")
-        return 1
+        return 2
     else:
         tm_dir = Path(benchmark_run_dir, "tm")
         try:
             tm_dir.mkdir()
             os.chdir(tm_dir)
         except Exception:
             logger.exception("failed to create the local tool meister directory")
-            return 1
+            return 3
     if not full_hostname or not hostname:
         logger.error(
             "ERROR - _pbench_hostname ('%s') and _pbench_full_hostname ('%s')"
             " environment variables are required",
             hostname,
             full_hostname,
         )
-        return 1
-    if os.environ.get("_PBENCH_UNIT_TESTS"):
-        # FIXME: this is an artifact of the unit test environment.
-        hostnames = "localhost"
+        return 4
+
+    # Determine the Tool Meister "hostname" to use for the Redis server to
+    # bind to, in addition to "localhost". That same host name will be used by
+    # the Tool Data Sink to bind to as well.
+    tm_bind_hostname = os.environ.get("PBENCH_TM_BIND_HOSTNAME", full_hostname)
+    hostnames_l = []
+    try:
+        localhost_ip = socket.gethostbyname("localhost")
+    except socket.error:
+        # Interesting networking environment, no IP address for "localhost" ...
+        localhost_ip = None
     else:
-        hostnames = f"localhost {full_hostname}"
+        # Add to the list of host names the Redis server will use.
+        hostnames_l.append("localhost")
+    try:
+        tm_bind_hostname_ip = socket.gethostbyname(tm_bind_hostname)
```

Comment:
I believe `gethostbyname("10.0.0.0")` does the right thing, but no harm being sure of that.

Reply:
This is actually trying to solve another problem.  On some systems `hostname -f` and/or `hostname` return different values from `localhost`, but still resolve to the same IP address as `localhost` (e.g. `localhost.localdomain`).  When that happens, Redis won't start because it tries to bind all the IP addresses given the list of host names but does not check to see if the IP address was already bound.

Reply:
OK